spec/support/shared_examples_for_fee_validators.rb

```diff
@@ -25,6 +30,33 @@
   end
 end
 
+RSpec.shared_examples 'common LGFS fee date govuk validations' do
+  describe '#validate_date' do
+    it { should_error_if_not_present(fee, :date, 'Enter the (.*?)(graduated) fee date') }
+
+    it 'adds error if too far in the past' do
+      fee.date = 11.years.ago
+      expect(fee).to_not be_valid
+      expect(fee.errors[:date]).to include(match(/(.*?)(Graduated) fee date cannot be too far in the past/))
+    end
+
+    it 'adds error if in the future' do
+      fee.date = 3.days.from_now
+      expect(fee).not_to be_valid
+      expect(fee.errors[:date]).to include(match(/(.*?)(Graduated) fee date cannot be too far in the future/))
+    end
+
+    it 'adds error if before the first repo order date' do
+      allow(claim).to receive(:earliest_representation_order_date).and_return(Date.today)
+      allow(fee).to receive(:claim).and_return(claim)
+
+      fee.date = Date.today - 3.days
+      expect(fee).not_to be_valid
+      expect(fee.errors[:date]).to include(match(/.* date must be no earlier than two years before .*/))
```

Comment:
As part of the re-writing the error messages, 'must be' needs replacing with  'cannot be'